tuf/client/client.go

```diff
@@ -111,10 +110,6 @@ func (c Client) checkRoot() error {
 		return fmt.Errorf("Cached root sha256 did not match snapshot root sha256")
```

Comment:
Oh, sorry, I misunderstood when I read this the first time.

We should fix filestore.go so we can test all the stores at a uniform interface level. @HuKeping I think without the fix in filestore.go (probably memorystore.go too), this change is unsafe.